tools/validate/validate.cpp

```diff
@@ -0,0 +1,205 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+#define _CRT_SECURE_NO_WARNINGS
+#include <algorithm>
+#include <array>
+#include <assert.h>
+#include <filesystem>
+#include <stdio.h>
+#include <stdlib.h>
+#include <string>
+#include <string_view>
+#include <utility>
+#include <vector>
+using namespace std;
+
+class BinaryFile {
+public:
+    explicit BinaryFile(const filesystem::path& filepath) {
+        m_file = _wfopen(filepath.c_str(), L"rb");
+
+        if (!m_file) {
+            fwprintf(stderr, L"Validation failed: %s couldn't be opened.\n", filepath.c_str());

```

Comment:
should we abort() after this?

Reply:
Not being able to open a file is bad (i.e. unexpected), but printing a message to stderr ensures that the build will fail. It doesn't necessarily preclude us from opening other files - we will continue iterating through the directory tree - so I think it's reasonable to proceed. Calling `abort()` could also make sense, but I was using that only for unrecoverable failures.